packages/mobile/src/screens/profile-screen/ProfileHeaderV2/SocialLink.tsx

```diff
@@ -0,0 +1,167 @@
+import { StyleProp, ViewStyle } from 'react-native'
+
+import IconInstagram from 'app/assets/images/iconInstagram.svg'
+import IconTikTok from 'app/assets/images/iconTikTokInverted.svg'
+import IconTwitterBird from 'app/assets/images/iconTwitterBird.svg'
+import {
+  Text,
+  LinkProps,
+  Link,
+  IconButtonProps,
+  Hyperlink
+} from 'app/components/core'
+import Skeleton from 'app/components/skeleton'
+import { makeStyles } from 'app/styles'
+import { EventNames } from 'app/types/analytics'
+import { make } from 'app/utils/analytics'
+import { useThemeColors } from 'app/utils/theme'
+
+import { useSelectProfile } from '../selectors'
+import { squashNewLines } from '../utils'
+
+const useStyles = makeStyles(({ palette, spacing, typography }) => ({
+  icon: {
+    height: 28,
+    width: 28,
+    fill: palette.neutral
+  },
+  skeleton: {
+    height: 28,
+    width: 28
+  },
+  withText: { flexDirection: 'row', alignItems: 'center' },
+  text: { marginLeft: spacing(2), marginBottom: 0 },
+  hyperlinkText: {
+    fontSize: typography.fontSize.medium,
+    color: palette.neutral
+  },
+  hyperlinkLink: {
+    fontSize: typography.fontSize.medium
+  }
+}))
+
+export type SocialLinkProps = LinkProps &
+  Pick<IconButtonProps, 'icon'> & {
+    style?: StyleProp<ViewStyle>
+    text?: string
+    showText?: boolean
+    hyperlink?: boolean
+  }
+
+export const SocialLink = (props: SocialLinkProps) => {
+  const { text, showText, url, icon: Icon, hyperlink, style, ...other } = props
+  const styles = useStyles()
+  const { neutral } = useThemeColors()
+
+  if (text === undefined) {
+    return <Skeleton style={styles.skeleton} />
+  }
+
+  if (text === null) {
+    return null
+  }
```

Comment:
This is kinda interesting - maybe worth an inline comment in the props delineating the difference btwn null and undefined?

Reply:
yeah that is a great callout, it's a convenience of the api that we can rely on undefined as 'LOADING' lol